SAMtags.tex

```diff
@@ -453,6 +455,106 @@ \subsubsection{Color space}
+stated base type to skip, stored as a delta to the last and starting
+with 0 as the first (or next) base.  Hence this number series is
+comparable to the numbers in an {\tt MD} tag.
+
+For example {\tt C+m,5,12,0;} tells us there are three 5-Methylcytosine
+bases in the original {\sf SEQ}.  The first 5 {\tt C} bases are
+unmodified and the 6th is modified, as are the 19th (12 inbetween the
+6th and 19th) and 20th.  Similarly {\tt G-m,14;} indicates the 15th
+{\tt G} is a 5-Methylcytosine on the opposite strand.
+
+This permits modifications to be listed on either strand with the rare
+potential for both strands to have a modification at the same site.
+
+If the modification is not one of the standard common types (listed
+below) it can be specified as a numeric ChEBI code.  For example
+{\tt C+76792,57;} is the same as {\tt C+h,57;}.
+
+An unmodified base of {\tt N} means count any base in {\sf SEQ}, not
+only those of {\tt N}.  Thus {\tt N+n,100;} means the 101st base is
+Xanthosine (n), irrespective of the sequence composition.
+
+The standard code types and their associated ChEBI values are listed
+below, taken from \emph{Modeling methyl-sensitive transcription factor
+motifs with an expanded epigenetic alphabet}, Coby Viner
+et.al. \url{https://www.biorxiv.org/content/10.1101/043794v1}.
+Additionally ambiguity codes {\tt A}, {\tt C}, {\tt G} and {\tt T}
+exist to represent unspecified modifications bases of their respective
+canonical base types, plus code {\tt N} to represent an unspecified
+modification of any base type.
+
+\begin{center}
+\begin{tabular}{lllll}
+{\bf Unmodified base} & {\bf Code} & {\bf Abbreviation} & {\bf Name} & {\bf ChEBI} \\
+\hline
+C & m & 5mC   & 5-Methylcytosine        & 27551 \\
+C & h & 5hmC  & 5-Hydroxymethylcytosine & 76792 \\
+C & f & 5fC   & 5-Formylcytosine        & 76794 \\
+C & c & 5caC  & 5-Carboxylcytosine      & 76793 \\
+C & C &       & Ambiguity code; any C mod & \\
+\hline
+T & g & 5hmU  & 5-Hydroxymethyluracil   & 16964 \\
+T & e & 5fU   & 5-Formyluracil          & 80961 \\
+T & b & 5caU  & 5-Carboxyluracil        & 17477 \\
+T & T &       & Ambiguity code; any T mod & \\
+\hline
+A & a & 6mA   & 6-Methyladenine         & 28871 \\
+A & A &       & Ambiguity code; any A mod & \\
+\hline
+G & o & 8oxoG & 8-Oxoguanine            & 44605 \\
+G & G &       & Ambiguity code; any G mod & \\
+\hline
+N & n & Xao   & Xanthosine              & 18107 \\
+N & N &       & Ambiguity code; any mod & \\
+\end{tabular}
+\end{center}
+
+\item[MP:Z:\tagvalue{qualities}]
+\hfill\\
+The {\tt MP} tag if present lists the Phred qualities of each
+modification listed in the {\tt MM} tag.  The length should match the
+number of position deltas from {\tt MM}.  The qualities are encoded in
+the same manner as the primary {\sf QUAL} field; one byte per quality
+with ASCII value Phred score + 33.  No separators should be present.
+
+For example {\tt MM:Z:C+m,5,12,3;C+h,57;} may have an associated
```

Comment:
would it be better to separate (e.g. with a semi-colon) the qualities that pertain to different modifications? in the example, the qualities would thus become `5EB;/`

Reply:
(perhaps with a `~` which has been used in the past to separate lists of qualities)

Reply:
If we have precedence elsewhere then it is good to follow convention.

Reviewing what we have already:
- OX/BZ.  OX is hyphen-concatenated sequence. BZ is defined to be the same length and BZ uses space to separate them (blergh, but non quality value hence chosen).
- RX/QX. As above
- CR/CY. As above
- BC/QT. As above

I've tried hard to expunge that period from my mind, but it seems we got to a consensus somewhere along the line. :-)

So possibly we should do the same here?  It's not quite analogous as we have a comma separated list instead of a sequence, but the same deal applies of choosing an invalid ASCII phred character as a separator.  We can't however claim the strings need to be equal length of course.

Thoughts anyone?  Space at least shows up really well when reading it.